syft/pkg/cataloger/java/tar_wrapped_archive_parser.go

```diff
@@ -0,0 +1,68 @@
+package java
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/anchore/syft/internal/file"
+
+	"github.com/anchore/syft/syft/artifact"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/pkg/cataloger/common"
+)
+
+// integrity check
+var _ common.ParserFn = parseTarWrappedJavaArchive
+
+var genericTarGlobs = []string{
+	"**/*.tar",
+	// gzipped tar
+	"**/*.tar.gz",
+	"**/*.tgz",
+	// bzip2
+	"**/*.tar.bz",
+	"**/*.tar.bz2",
+	"**/*.tbz",
+	"**/*.tbz2",
+	// brotli
+	"**/*.tar.br",
+	"**/*.tbr",
+	// lz4
+	"**/*.tar.lz4",
+	"**/*.tlz4",
+	// sz
+	"**/*.tar.sz",
+	"**/*.tsz",
+	// xz
+	"**/*.tar.xz",
+	"**/*.txz",
+	// zst
+	"**/*.tar.zst",
+}
+
+// TODO: when the generic archive cataloger is implemented, this should be removed (https://github.com/anchore/syft/issues/246)
```

Comment:
So rather than hook these as functions into glob matching under the java cataloger you want a generic archive cataloger that can then delegate discovered packages to their respective languages?

Reply:
TBD, but that is one possible approach. It may be that the `source.FileResolver` implementations take care of this detail for us.